packages/flutter/lib/src/foundation/debug.dart

```diff
@@ -8,6 +8,16 @@ import 'assertions.dart';
 import 'platform.dart';
 import 'print.dart';
 
+/// A constant that is true if the application was compiled in release mode.
+///
+/// More specifically, this is a constant that is true if the application was
```

Comment:
/cc @rmacnak-google for the exact definition of `dart.vm.product`.

Reply:
Sorry, it should have been `-Ddart.vm.product`

I've switch it to be a new flag, set by the Flutter build tool: `-Dflutter.buildMode.release` (and profile, and debug), so that it's more rational, and matches better what Flutter means by release, etc.